.github/workflows/pr.yml

```diff
@@ -235,56 +227,31 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, macos-latest]
-        dotnet-version: [8.0.x]
+        dotnet-version: [8.0]
     runs-on: ${{ matrix.os }}
     steps:
       - name: Fetch secrets from ESC
         id: esc-secrets
         continue-on-error: true
         uses: pulumi/esc-action@cf5b30703ffd5ad60cc3a880c09b3a9592b9372d # v1
       - name: Set TARGET_FRAMEWORK
-        shell: bash
-        run: |
-          if [[ "${{ matrix.dotnet-version }}" == "8.0.x" ]]; then
-            echo "TARGET_FRAMEWORK=net8.0" >> $GITHUB_ENV
-          elif [[ "${{ matrix.dotnet-version }}" == "9.0.x" ]]; then
-            echo "TARGET_FRAMEWORK=net9.0" >> $GITHUB_ENV
-          else
-            echo "Unexpected dotnet-version: ${{ matrix.dotnet-version }}"
-            exit 1
-          fi
+        run: echo "TARGET_FRAMEWORK=net${{ matrix.dotnet-version }}" >> $GITHUB_ENV
       - name: Checkout code
         uses: actions/checkout@v5
         with:
           submodules: recursive
-      - name: Setup dotnet SDK
-        id: setup-dotnet
-        uses: actions/setup-dotnet@v4
-        with:
-          dotnet-version: ${{ matrix.dotnet-version }}
-      - name: Create global.json
-        # This ensures that we use the matrix version instead of the runner's default version
-        # https://github.com/actions/setup-dotnet?tab=readme-ov-file#matrix-testing
-        run: |
-          echo '{"sdk":{"version": "${{ steps.setup-dotnet.outputs.dotnet-version }}"}}' > ./global.json
-      - name: Dotnet version
-        run: dotnet --version
-      - name: Set up Go 1.24.x
-        uses: actions/setup-go@v5
-        with:
-          go-version: 1.24.x
+      - name: Setup mise and install tools
+        uses: jdx/mise-action@v3.3.1
+      - name: Set dotnet version from matrix
+        run: mise use dotnet@${{ matrix.dotnet-version }}
       - name: Install Pulumi CLI
-        uses: pulumi/actions@d7ceb0215da5a14ec84f50b703365ddf0194a9c8 # v6
+        uses: pulumi/actions@v6
```

Comment:
If we want to exercise the `pulumi/actions` here, there's no point pinning a specific version of V6

Reply:
Yeah, imo pinning actions to a specific version is an antipattern anyway. If nobody is gonna actually check the code there's not much point in pinning anything (and we own actions, so there's definitely no issues there)